File: java/src/main/java/br/com/emendes/problems/LengthOfLastWord.java
package br.com.emendes.problems;

/**
 * Dado uma String s que consiste de palavras e espaços, retorne o comprimento da última palavra da String s.<br><br>
 *
 * Uma palavra é uma substring que consiste apenas caracteres que não são espaço.<br><br>
 *
 * Restrições:<br>
 * 1 <= length <= 10⁴<br>
 * s consiste apenas de letras inglesas e espaços ' '.<br>
 * Existe pelo menos uma palavra em s.
 */
public class LengthOfLastWord {

  public int lengthOfLastWord(String s) {
    int length = s.length();
    int index = length - 1;

    while (s.charAt(index) == ' ') {
      index--;
      length--;
    }

    while (index >= 0 && s.charAt(index) != ' ') {
      index--;
    }

    return length - (index + 1);
  }

}

File: javascript/problems/LengthOfLastWord.js
/**
 * Dado uma string s que consiste de letras e espaços em branco, retorne o
 * comprimento da última palavra da string.
 *
 * Uma palavra é uma substring contendo caracteres sem espaço em branco.
 *
 * Restrições:
 *
 * 1 <= s.length <= 10⁴
 *
 * s consiste de letras e espaços em branco ' '.
 *
 * Existe ao menos uma palavra em s.
 *
 * @param {string} s
 * @return {number}
 */
const lengthOfLastWord = function (s) {
  let index = s.length - 1;
  let lengthLastWord = 0;
  while (index >= 0 && s[index] === " ") index--;
  while (index >= 0 && s[index] !== " ") {
    index--;
    lengthLastWord++;
  }

  return lengthLastWord;
};
